### ISSUE TITLE
Failed to compile with go1.4.1

### ISSUE THREAD

**Issue by stamp** [EconomistDigitalSolutions/ramlapi#4]
Hi

I think the ramlapi.ResourceVerbs method from github.com/EconomistDigitalSolutions/ramlapi have changed return types and causing the compile to fail.

I suppose something like len() == 0 would work on line 79 and 105 but the other two need some more work

```
# github.com/EconomistDigitalSolutions/ramlapi/ramlgen
./main.go:79: invalid operation: n == "" (mismatched types map[string]string and string)
./main.go:82: cannot use n (type map[string]string) as type string in field value
./main.go:105: invalid operation: n == "" (mismatched types map[string]string and string)
./main.go:108: cannot use n (type map[string]string) as type string in field value
```

**Comment by rlhatcher** [comment]
Can you let me know the RAML you're using? That will help immensely.

**Comment by stamp** [comment]
Sorry, but I haven`t created a RAML file yet. But it shouldn`t matter because I haven´t been able to compile the ramlgen tool yet. The RAML file has nothing to do with the compilation of the ramlgen tool.

**Comment by buddhamagnet** [comment]
Thanks @stamp - I'm in Greece right now but will be back in the UK Sep 6 and will have a look see then, and in the meantime the mental giant that is @rlhatcher can help also. Thanks!

**Comment by buddhamagnet** [comment]
Hey @stamp so yep looks like the issue is keeping ramlgen in step with API changes in ramlapi proper, thanks for the spot. Could you try and compile with this branch: 

https://github.com/EconomistDigitalSolutions/ramlapi/tree/ISSUE-4-ramlgen-disparity

If that's good I'll merge into master.

**Comment by buddhamagnet** [comment]
@stamp and we'll roll some tests into ramlgen to make sure it keeps in step with any breaking changes in the upstream API.

**Comment by stamp** [comment]
@buddhamagnet 

Nice!

I have tested it and it compiles just fine. Now I will start creating my RAML file and test your tool :)

Proof of success:

```
~/go/src/github.com/EconomistDigitalSolutions/ramlapi/ramlgen/ramlgen 
2015/08/26 09:58:19 Processing API spec for api.raml
2015/08/26 09:58:20 Created handlers in  handlers_gen.go
```

**Comment by stamp** [comment]
We using travis for checking stuff like that

Take a look at our travis.yml: https://github.com/stampzilla/stampzilla-go/blob/master/.travis.yml

Especially the line 

```
go test -v `go list ./...`
```

**Comment by buddhamagnet** [comment]
@stamp @rlhatcher merged and tagged at 0.3.1. Thanks!
